services/spar/src/Spar/App.hs

```diff
@@ -174,19 +184,19 @@ instance SPStoreID Assertion (Spar r) where
   unStoreID r = wrapMonadClient $ Data.unStoreAssID r
   isAliveID r = wrapMonadClient $ Data.isAliveAssID r
 
-instance SPStoreIdP SparError (Spar r) where
+instance Member IdPEffect.IdP r => SPStoreIdP SparError (Spar r) where
   type IdPConfigExtra (Spar r) = WireIdP
   type IdPConfigSPId (Spar r) = TeamId
 
   storeIdPConfig :: IdP -> Spar r ()
-  storeIdPConfig idp = wrapMonadClient $ Data.storeIdPConfig idp
+  storeIdPConfig idp = wrapMonadClientSem $ IdPEffect.storeConfig idp
 
   getIdPConfig :: IdPId -> Spar r IdP
-  getIdPConfig = (>>= maybe (throwSpar (SparIdPNotFound mempty)) pure) . wrapMonadClientWithEnv . Data.getIdPConfig
+  getIdPConfig = (>>= maybe (throwSpar (SparIdPNotFound mempty)) pure) . wrapMonadClientSem . IdPEffect.getConfig
```

Comment:
As best as I can tell, `wrapMonadClientWithEnv` does nothing here. Can someone confirm?

Reply:
Correct, good catch!  (And it's really annoying to reason that out, because of `saml2-web-sso`'s type classes that unnecessarily abstract away all traces of related evidence...)